### PR TITLE
SAIC-787 Little speed up for cinder volume migration

### PR DESCRIPTION
**Pull request by aguzikova**
Add some caching as a temporarily solution to speed up a migration a
little bit.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.storage.cinder_database
cloudferrylib/os/storage/cinder_database.py:232: [W0125(using-constant-test), CinderStorage._filter_quotas_list] Using a conditional statement with a constant value
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:162: [W0212(protected-access), GlanceImage.get_ref_image] Access to a protected member _resp of a client class
cloudferrylib/os/image/glance_image.py:246: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:287: [W1202(logging-format-interpolation), GlanceImage.make_image_info] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:298: [W1202(logging-format-interpolation), GlanceImage.make_image_info] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:301: [W1202(logging-format-interpolation), GlanceImage.make_image_info] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:311: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/image/glance_image.py:359: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:365: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:374: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:377: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:397: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:424: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:451: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions and pass the % parameters as arguments

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1787/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.storage.cinder_database
cloudferrylib/os/storage/cinder_database.py:232: [W0125(using-constant-test), CinderStorage._filter_quotas_list] Using a conditional statement with a constant value
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:219: [W0212(protected-access), GlanceImage.get_ref_image] Access to a protected member _resp of a client class
cloudferrylib/os/image/glance_image.py:303: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:367: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method
************* Module tests.cloudferrylib.os.image.test_glance_image
tests/cloudferrylib/os/image/test_glance_image.py:50: [W0212(protected-access), GlanceImageTestCase.setUp] Access to a protected member _resp of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.storage.cinder_database
cloudferrylib/os/storage/cinder_database.py:232: [W0125(using-constant-test), CinderStorage._filter_quotas_list] Using a conditional statement with a constant value
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:219: [W0212(protected-access), GlanceImage.get_ref_image] Access to a protected member _resp of a client class
cloudferrylib/os/image/glance_image.py:303: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:367: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method
************* Module tests.cloudferrylib.os.image.test_glance_image
tests/cloudferrylib/os/image/test_glance_image.py:56: [W0212(protected-access), GlanceImageTestCase.setUp] Access to a protected member _resp of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/image/glance_image.py [hunk 1]
```diff
@@ -41,16 +41,17 @@
...
-
-    """
+    """ The main class for working with Openstack Glance Image Service. """
 
     def __init__(self, config, cloud):
         self.config = config
         self.host = config.cloud.host
         self.cloud = cloud
         self.identity_client = cloud.resources['identity']
+        self.tn_id_to_name = {
```
Comment (roman-verchikov) [inline cloudferrylib/os/image/glance_image.py]: Use `cloudferrylib.utils.cache.Cached` instead

**Comment by archyufa** [comment]
Will be good to try tomorrow in staging and prod

**Pull request closed without merging**